lib/dk/dry_runner.rb

```diff
@@ -1,11 +1,33 @@
 require 'dk/config_runner'
+require 'dk/has_the_stubs'
 
 module Dk
 
   class DryRunner < ConfigRunner
+    include HasTheStubs
 
     # run with disabled cmds, just log actions, but run all sub-tasks
 
+    private
+
+    def has_the_stubs_build_local_cmd(cmd_str, given_opts)
+      given_opts ||= {}
+      if given_opts[:dry_tree_run]
+        Local::Cmd.new(cmd_str, given_opts)
+      else
+        Local::CmdSpy.new(cmd_str, given_opts)
+      end
```

Comment:
@jcredding maybe less lines to do it this way (both here and on the remote version)?  Feel free to ignore this also.

```ruby
cmd_klass = given_opts[:dry_tree_run] ? Local::Cmd : Local::CmdSpy
cmd_klass.new(cmd_str, given_opts)
```

One downside is that it couples the init signature of both classes.  It does remove the init args duplication tho.  Anyway, I'm cool with this either way.